core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostEstimator.scala

```diff
@@ -73,14 +85,21 @@ class DefaultCostEstimator(
     case Some(_) =>
   }
 
-  override def estimate(region: Region, resourceUnits: Int): Double = {
-    this.operatorEstimatedTimeOption match {
+  override def allocateResourcesAndEstimateCost(
+      region: Region,
+      resourceUnits: Int
+  ): (Region, Double) = {
+    // Currently the dummy cost from resourceAllocator is discarded.
+    val (newRegion, _) = resourceAllocator.allocate(region)
```

Comment:
I am confused with the role of `resource allocator` and `cost estimator`. if resource allocator is also returning a cost, why do we need another cost estimator? are we going to stop letting resource allocator to emit a cost?

Reply:
Resource allocator is currently only returning a dummy cost. The current cost estimator is estimating cost independent of the resource allocator. In the future when resource allocator can indeed give a meaningful cost, we can use that cost instead. This PR is merging the two together for future benefits and for clearer representation in the scheduler diagram.

Reply:
when we use the estimated cost from the resource allocator, what does cost estimator do? is it just a wrapper?

Reply:
We can keep only resource allocator in the future if it can both allocate resources and estimate costs. For now we are doing this refactoring so that the two are not separated (to align with the diagram).